java/src/main/java/com/google/oak/transparency/SignatureVerifier.java

```diff
@@ -49,24 +61,51 @@ private static Optional<Failure> failure(String message) {
     /**
      * Verifies the signature over content bytes using a public key.
      *
-     * @param signature      The base64-encoded signature
-     * @param publicKeyBytes The PEM-encoded public key
-     * @param contentBytes   The serialized content
-     * @return Empty if the verification succeeds, or a failure otherwise
+     * @param signatureBytes the raw ECDSA signature, likely 71 bytes long
+     * @param publicKeyBytes the raw public key
+     * @param contentBytes   the serialized content
+     * @return empty if the verification succeeds, or a failure otherwise
      */
     public static Optional<Failure> verify(
-            String signature, byte[] publicKeyBytes, byte[] contentBytes) {
-        if (signature == null || signature.length() == 0) {
+            byte[] signatureBytes, byte[] publicKeyBytes, byte[] contentBytes) {
+        if (signatureBytes == null || signatureBytes.length == 0) {
             return failure("empty signature");
         }
+        if (publicKeyBytes == null || publicKeyBytes.length == 0) {
+            return failure("empty public key");
+        }
         if (contentBytes == null || contentBytes.length == 0) {
             return failure("empty content");
         }
-        if (publicKeyBytes == null || publicKeyBytes.length == 0) {
-            return failure("empty public key");
+
+        X509EncodedKeySpec keySpec = new X509EncodedKeySpec(publicKeyBytes);
+        Exception exception = null;
+        boolean outcome = false;
```

Comment:
Given that the return types are already not always semantically aligned (e.g. sometimes option.empty means success, sometimes failure), I suggest using a clearer name for this variable, e.g. `success`

Reply:
Right, done.